_episodes/08-connected-components.md

```diff
@@ -261,156 +319,388 @@ max: 11
+should be of a certain size or area. So we could use a minimum area as
+a criterion for when an object should be detected. To apply such a
+criterion, we need a way to calculate the area of objects found by
+connected components. Recall how we determined the root mass in the
+[Thresholding]({{ page.root }}/07-thresholding) by counting the pixels
+in the binary mask. But here we want to calculate the area of several
+objects in the labeled image. The skimage library provides the
+function `skimage.measure.regionprops` to measure the properties of
+labeled regions. It returns a list of `RegionProperties` that describe
+each connected region in the images. The properties can be accessed
+using the attributes of the `RegionProperties` data type. Here we will
+use the properties `"area"` and `"label"`. You can explore the skimage
+documentation to learn about other properties available.
+
+We can get a list of areas of the labeled objects as follows:
 
-> ## Plot a histogram of the object area distribution (15 min)
->
-> In the previous exercise we wrote the `CCA-count.py` program and explored how the object count changed with the parameters.
-> We had a hard time making the script print out the right number of objects.
-> In order to get closer to a solution to this problem, we want to look at the distribution of the object areas.
-> Calculate the object properties using `skimage.measure.regionprops` and [generate a histogram]({{ page.root }}/05-creating-histograms).
->
-> Make a copy of the `CCA.py` script and modify it to also produce a plot of the histogram of the object area.
+~~~
+# compute object features and extract object areas
+object_features = skimage.measure.regionprops(labeled_image)
+object_areas = [objf["area"] for objf in object_features]
+object_areas
+~~~
+{: .language-python}
+
+This will produce the output
+~~~
+[318542, 1, 523204, 496613, 517331, 143, 256215, 1, 68, 338784, 265755]
+~~~
+{: .output}
+
+> ## Plot a histogram of the object area distribution (10 min)
 >
-> What does the histogram tell you about the objects?
+> Similar to how we determined a "good" threshold in the
+> [Thresholding]({{ page.root }}/07-thresholding) episode, it is often
+> helpful to inspect the histogram of an object property. For example,
+> we want to look at the distribution of the object areas.
 >
-> Hint: Try to generate a list of object areas first.
+> 1. Create and examine a [histogram]({{ page.root }}/05-creating-histograms)
+> of the object areas obtained with `skimage.measure.regionprops`.
+> 2. What does the histogram tell you about the objects?
 >
 > > ## Solution
 > >
+> > The histogram can be plotted with
 > > ~~~
-> > # add the import for pyplot
-> > from matplotlib import pyplot as plt
-> >
-> > # compute object features and extrac object areas
-> > object_features = skimage.measure.regionprops(labeled_image)
-> > object_areas = [objf["area"] for objf in object_features]
 > > plt.hist(object_areas)
```

Comment:
It might be good to add axis labels to the histogram here? It just makes it easier to understand at a glance.

Reply:
Thanks for the suggestion @K-Meech. I have added the axes labels in 942f05464602218a2e09cd9edd1ef3cae0b591a0.